scrub_hub_project/authenticate/views.py

```diff
@@ -49,30 +51,53 @@ def dashboard_view(request):
     return render(request, 'authenticate/dashboard.html')
 
 @ensure_csrf_cookie
+@api_view(['GET', 'POST']) # Required to have request.data
 def register_view(request):
     if request.method == "POST":
-        data = json.loads(request.body)
-        email = data.get("email")
-        password = data.get("password")
-        confirm_password = data.get("confirm_password")
-        first_name = data.get("first_name")
-        last_name = data.get("last_name")
-        phone_number = data.get("phone_number")
-        employee_id = data.get("employee_id")
-        registration_code = data.get("registration_code")
-
-        if not all([email, password, confirm_password, first_name, last_name, phone_number, employee_id, registration_code]):
-            return JsonResponse({"detail": "Missing information"}, status=400)
+        data: dict = {}
+        expected_fields = [
+            'first_name', 'last_name',
+            'password', 'confirm_password',
+            'phone_number', 'email',
+            'employee_id', 'registration_code',
+            'public_key', 'private_key',
+        ]
+        for field in expected_fields:
+            data[field] = request.data.get(field)
+            if data[field] is None:
+                return JsonResponse({"detail": "Missing information"}, status=400)
+            if type(data[field]) == InMemoryUploadedFile:
+                data[field] = data[field].read()
 
-        if password != confirm_password:
+        if data['password'] != data['confirm_password']:
             return JsonResponse({"detail": "Passwords do not match!"}, status=400)
-        
-        if CustomUser.objects.filter(email=email).exists():
+        del data['confirm_password']
+
+        if CustomUser.objects.filter(email=data['email']).exists():
             return JsonResponse({"detail": "Username already taken"}, status=400)
 
-        user = CustomUser.objects.create_user(email=email, password=password, first_name=first_name, last_name=last_name, phone_number=phone_number, employee_id=employee_id, registration_code=registration_code)
+        user = CustomUser.objects.create_user(**data)
         user.save()
 
         return JsonResponse({"detail": "User successfully registered"}, status=201)
     elif request.method == "GET":
         return render(request, 'authenticate/register.html')
+
+@api_view(['POST'])
+def set_keys(request):
+    user: CustomUser = request.user
+    if not user.is_authenticated:
+        return JsonResponse({"detail": "You are not logged in!"}, status=400)
+    data: dict = {}
+    expected_fields = [ 'public_key', 'private_key' ]
+    for field in expected_fields:
+        data[field] = request.data.get(field)
+        if data[field] is None:
+            return JsonResponse({"detail": "Missing information"}, status=400)
+        if type(data[field]) == InMemoryUploadedFile:
```

Comment:
I don't understand what you're asking. This is server-side code, so it can't even access local storage.

set_keys is an API endpoint called when a user logs in but doesn't yet have public/private keys. The client then creates them. See scrub_hub_vite/src/components/Login.tsx lines 34-51.

Reply:
Your code is not clear what is happening, there should be more commenting or better quality of code such that others can actually follow what is going on in the project